src/main/java/sg/christopher/duke/Duke.java

```diff
@@ -0,0 +1,284 @@
+package sg.christopher.duke;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.stage.Stage;
+
+import sg.christopher.duke.entities.Deadline;
+import sg.christopher.duke.entities.Event;
+import sg.christopher.duke.entities.Task;
+import sg.christopher.duke.entities.Todo;
+import sg.christopher.duke.io.DataManager;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+public class Duke extends Application {
+    private static List<Task> savedItems = loadSavedItems();
+
+    /**
+     * Hydrates the task data store with saved data from disk.
+     *
+     * @return list of tasks that are saved on disk
+     */
+    public static List<Task> loadSavedItems() {
+        List<Task> saved = DataManager.readList();
+        return saved != null ? saved : new ArrayList<>();
+    }
+
+    /**
+     * Saves a task to memory and writes it to disk.
+     *
+     * @param task the task to be saved
+     */
+    public static void saveItem(Task task) {
+        savedItems.add(task);
+        DataManager.writeList(savedItems);
+    }
+
+    /**
+     * Loads saved tasks from disk, and get a specific task using its index.
+     *
+     * @param index index of the task to be retrieved
+     * @return the task at that index
+     */
+    public static Task loadItem(int index) {
+        savedItems = DataManager.readList();
+        return savedItems.get(index);
+    }
+
+    private static void doneHandler(String userInput) {
+        int taskNo;
+        try {
+            taskNo = Integer.parseInt(userInput.split(" ")[1]);
+        } catch (ArrayIndexOutOfBoundsException aioobe) {
```

Comment:
Just a small suggestion, perhaps you can consider using a single letter 'e' to represent the exception but taking the first letter of every word is an interesting way to do it as well!